grpc-api/src/main/java/io/deephaven/grpc_api/browserstreaming/BrowserStream.java

```diff
@@ -0,0 +1,263 @@
+                }
+                boolean queueMsg = false;
+                if (processingMessage) {
+                    queueMsg = true;
+                    log.debug().append(logIdentity).append("queueing; next seq=").append(nextSeq)
+                            .append(" recv seq=").append(streamData.getSequence()).endl();
+                } else if (streamData.getSequence() != nextSeq) {
+                    queueMsg = true;
+                    log.debug().append(logIdentity).append("queueing; waiting seq=").append(nextSeq)
+                            .append(" recv seq=").append(streamData.getSequence()).endl();
+                }
+                if (queueMsg) {
+                    if (pendingSeq == null) {
+                        pendingSeq = new RAPriQueue<>(1, MessageInfoQueueAdapter.getInstance(), Message.class);
+                    }
+                    pendingSeq.enter(new Message<>(message, streamData));
+                    return;
+                }
+            } else { // Mode.MOST_RECENT
+                if (streamData.getSequence() < nextSeq
+                        || (streamData.getSequence() == nextSeq && processingMessage) // checks for duplicate
+                        || (queuedStreamData != null && streamData.getSequence() < queuedStreamData.getSequence())) {
+                    // this message is too old
+                    log.debug().append(logIdentity).append("dropping; next seq=").append(nextSeq)
+                            .append(" queued seq=").append(queuedStreamData != null ? queuedStreamData.getSequence() : -1)
+                            .append(" recv seq=").append(streamData.getSequence()).endl();
+                    return;
+                }
+                // is most recent msg seen
+                if (processingMessage) {
+                    log.debug().append(logIdentity).append("queueing; processing seq=").append(nextSeq)
+                            .append(" recv seq=").append(streamData.getSequence()).endl();
+                    queuedStreamData = streamData;
+                    queuedMessage = message;
+                    return;
+                }
+            }
+
+            nextSeq = streamData.getSequence() + 1;
+            processingMessage = true;
+        }
+
+        do {
+            synchronized (this) {
+                if (streamData.isHalfClose()) {
+                    onComplete();
+                    processingMessage = false;
+                    return;
+                }
+            }
+            try {
+                marshaller.onMessageReceived(message);
+            } catch (final RuntimeException e) {
+                onError(e);
+                return;
+            }
+
+            synchronized (this) {
+                if (mode == Mode.IN_ORDER) {
+                    message = pendingSeq == null ? null : pendingSeq.top().getMessage();
```

Comment:
As discussed this NPEs; and you wanted to make it pretty again. Maybe stretching the logic out like this will make it clean again:

```java
if (pendingSeq == null) {
  break;
}
Message<T> topOfQueue = pendingSeq.top();
if (topOfQueue == null) {
  break;
}
// etc
```

Reply:
LMK what you think of the patch I put up compared to your suggested edit above